driver/ppm_events.c

```diff
@@ -1258,6 +1261,9 @@ int32_t compat_parse_readv_writev_bufs(struct event_filler_arguments *args, cons
 
 	copylen = iovcnt * sizeof(struct compat_iovec);
 
+	if (unlikely(iovcnt >= 0xffffffff))
+		return PPM_FAILURE_BUFFER_FULL;
+
```

Comment:
Checking this _after_ calculating `copylen` feels wrong to me but should be safe (as far as I can tell, we don't trigger any UB even with overflow, since we're only multiplying unsigned values).

Reply:
I agree that moving this before copylen is calculated would be better on a logic point of view but as it is now the checks loris added are still avoiding bad things to happen in case anything happens to iovcnt so.